pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -355,6 +359,15 @@ func ConfigFromEnv() Config {
 	cfg.Global.ApplicationCredentialName = os.Getenv("OS_APPLICATION_CREDENTIAL_NAME")
 	cfg.Global.ApplicationCredentialSecret = os.Getenv("OS_APPLICATION_CREDENTIAL_SECRET")
 
+	endpointType := os.Getenv("OS_ENDPOINT_TYPE")
+	if endpointType == "internal" || endpointType == "internalURL" {
+		cfg.Global.EndpointType = gophercloud.AvailabilityInternal
+	} else if endpointType == "admin" || endpointType == "adminURL" {
+		cfg.Global.EndpointType = gophercloud.AvailabilityAdmin
```

Comment:
can we just use Type Conversion here?
Eg:
cfg.Global.EndpointType = Availability(cfg.Global.EndpointType)

Reply:
And you are using this conversion only for env variables not for config file ones.. why?

Reply:
> can we just use Type Conversion here?

Unfortunately no, since we have to make sure that the value is one of the supported.

> And you are using this conversion only for env variables not for config file ones.. why?

You're right, this requires custom unmarshaller with additional checks, or using the conversion func inside every dependent package. Probably it is better to create a conversion func and use the `string` type for the `EndpointType`.